### PR TITLE
For #22851  Add logo animation for wallpapers

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by MatthewTighe — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -1200,6 +1222,17 @@ class HomeFragment : Fragment() {
...
+    // We want to show the animation in a time when the user less distracted
+    // The Heuristics are:
+    // 1) The animation hasn't shown before.
+    // 2) The user has onboarded.
+    // 3) It's the third time the user enters the app.
+    @Suppress("MagicNumber")
+    private fun shouldAnimateLogoForWallpaper() =
+        shouldEnableWallpaper() && context?.settings()?.shouldAnimateFirefoxLogo == true
+            && onboarding.userHasBeenOnboarded()
+            && context?.settings()?.numberOfAppLaunches == 3
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:1234]: @MatthewTighe just to make you aware, I added some extra logic, after the product input. They want to show the animation when the user is less distracted with other parts of the app, specifically after onboarding has been finished.

Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:1234]: Thanks for the update! Looks good to me

**Comment by Amejia481** [comment]
👋🏽 
@mcarare @MarcLeclair. I'm having a hard time trying to figure, what it's causing [verifyRunBlockingAndStrictModeSuppresionCount](https://console.firebase.google.com/u/1/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8623599678080407412/executions/bs.8744e3ea1286c83c/test-cases) to failed. If it's not much trouble would you mind giving me a hand :) .

I don't understand how I'm increasing the `suppressionCount`, without using `resetAfter`.
I also tested removing the two `@Suppress("MagicNumber")` locally and I still get the failing test.

```
java.lang.AssertionError: StrictMode suppression count does not match expected count.

    If this PR removed a StrictMode suppression call, great! Please decrease the count.

    Did this PR add or call code that increases the StrictMode suppression count?
    Did you know that suppressing a StrictMode violation can introduce performance regressions?
    Please do your best to implement a solution without adding StrictMode suppression calls.
    Please consult the perf team if you have questions or believe that having this call
    is the optimal solution.


 expected:<20> but was:<21>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.mozilla.fenix.perf.StartupExcessiveResourceUseTest.verifyRunBlockingAndStrictModeSuppresionCount(StartupExcessiveResourceUseTest.kt:88)
	at java.lang.reflect.Method.invoke(Native Method)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
	at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
	at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
    
```

**Comment by Amejia481** [comment]
I found the reason of the test regression
It's because [onboarding.userHasBeenOnboarded() ](https://github.com/mozilla-mobile/fenix/pull/23311/files#diff-41731714e97083a1b73584da2837ab11c22796c25b892b1e86ce4190ebba2633R1234). In this case, it's a product requirement that we make sure this animation show after the user has finished onboarding. What we normally do on this cases  @mcarare @MarcLeclair. Should I alter the based line in the test ?

**Comment by MarcLeclair** [comment]
@Amejia481  So, if there's no way around us calling this in the background on startup and since it's only for onboarding, I don't think this is going to be a big issue. However, if we want to keep things consistent with other parts of the code that required removing `StrictMode` to work, we can use [`StrictModeManager.resetAfter()`](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/perf/StrictModeManager.kt#L113).  It shouldn't be a big change to the code and should make it pass I think

**Comment by Amejia481** [comment]
> @Amejia481 So, if there's no way around us calling this in the background on startup and since it's only for onboarding, I don't think this is going to be a big issue. However, if we want to keep things consistent with other parts of the code that required removing `StrictMode` to work, we can use [`StrictModeManager.resetAfter()`](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/perf/StrictModeManager.kt#L113). It shouldn't be a big change to the code and should make it pass I think

Awesome, thanks I'll give it a try to adding `StrictModeManager.resetAfter()`

**Pull request merged**